### PR TITLE
fix virtual product no uri error

### PR DESCRIPTION
**Pull request by uchchwhash**
### Reason for this pull request
Virtual product load mechanism does not ensure that the datasets have location. The usual `dc.load` does.

### Proposed changes
Throw away datasets with no location.

### CODE REVIEW

**Comment by Kirill888** [comment]
@uchchwhash please don't merge until after tests are merged in my WIP PR

**Review by bellemae — APPROVED** [review]
Fantastic, thanks!

**Pull request merged**
